app/controllers/op_turbo_sandbox_controller.rb

```diff
@@ -0,0 +1,133 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class OpTurboSandboxController < ApplicationController
+  include OpTurbo::ComponentStream
+
+  before_action :authorize_global
+  before_action :set_projects
+
+  def index; end
+
+  def edit
+    @project = Project.find(params[:id])
+
+    update_via_turbo_stream(
+      component: OpTurboSandbox::Projects::InlineEditComponent.new(project: @project, state: :edit)
+    )
+
+    respond_to_with_turbo_streams
+  end
+
+  def create
+    @project = Project.new(project_params)
+
+    if @project.save
+      # Option 1:
+      # update the whole list, which will include the new project
+      # update_via_turbo_stream(
+      #   component: OpTurboSandbox::Projects::IndexComponent.new(projects: @projects)
+      # )
+
+      # Option 2:
+      # append (or prepend) the new project to the list
+      append_via_turbo_stream(
+        component: OpTurboSandbox::Projects::InlineEditComponent.new(project: @project),
+        target_component: OpTurboSandbox::Projects::IndexComponent.new(projects: @projects)
+      )
+
+      # reset the form
+      update_via_turbo_stream(
+        component: OpTurboSandbox::Projects::NewComponent.new(project: Project.new)
+      )
+    else
+      # update the form with errors
+      update_via_turbo_stream(
+        component: OpTurboSandbox::Projects::NewComponent.new(project: @project)
+      )
+    end
+
+    respond_to_with_turbo_streams
+  end
```

Comment:
Especially if we are using the actions also for non turbo_stream responses (which I like), this kind of writing looks wasteful to me. 

The append/update_via_turbo_stream methods are called regardless of whether we are actually rendering a stream response in the end or not. 

This problem could be removed by turning the `modify_via_turbo_streams` into something like 
```ruby
def modify_via_turbo_stream(component:, action:)
  return unless formats.any?(:turbo_stream) # or something like that. Didn't check the exact value.

  @turbo_streams << component.render_as_turbo_stream(
        view_context:,
        action:
      )
end
```

But even with that, a real live example of the method above used for multiple formats would look differently from what is used here: 
* There would be a redirect in case the saving fails (html only)
* There would be some flash message in case the saving fails (html only or another turbo stream update)
* There might be a flash message in case the saving succeeds (html only or another turbo stream update)
So such a controller would become rather complicated to read. 

And even without it, there are quite a bunch of updates/append statements within this simplified example. Thinking about a method that updates the work package table view when a filter is added, there will be probably way more modifications that need to take place (adding the filter to the list of selected filters, reseting the filter selector to have the option removed, updating the table (but keeping the columns selected), closing the split screen as the work package displayed in there is no longer part of the result set ...)

The idea I am pondering is, similar to having a full template rendered on an html request, have another (single) object that is rendered when the controller method returns. 

It could be triggered by having a 

```ruby
respond_to do |format|
 format.html { render }
 format.turbo_stream { Projects::CreateViewObject.new([the parameters necessary] }
end
```

That `Projects::CreateViewObject` (not set on the name) would then orchestrate all the changes necessary to the page. 

That would free the controller of needing to know how to update the page where it mingles with the other responsibilities (parsing params, executing an action, choosing the response). 

The triggering could also be done implicitly, same as how it is done for rendering the view named accordingly for an html request.

That idea of mine is unfinished and has a lot of rough edges, so feedback is welcomed.